client/src/utils/CSSHelp.ts

```diff
@@ -0,0 +1,115 @@
+  // Could use some clean up...
+  // @ts-ignore
+  const cssStyleRules = Array.from(this.parentStyleSheet.cssRules)?.filter(
+    ele => ele.type === CSSTypes.style
+  ) as CSSStyleRule[];
+  const previousStyleRule = cssStyleRules.find(
+    ele => ele?.selectorText === selector
+  );
+  // @ts-ignore
+  const currPosition = Array.from(this.parentStyleSheet.cssRules).indexOf(this);
+  // @ts-ignore
+  const prevPosition = Array.from(
+    // @ts-ignore
+    previousStyleRule?.parentStyleSheet?.cssRules
+    // @ts-ignore
+  ).indexOf(previousStyleRule);
+  return currPosition > prevPosition;
+};
+
+class CSSHelp {
+  doc: HTMLDocument;
+  constructor(doc: HTMLDocument) {
+    this.doc = doc;
+  }
+  getStyleDeclarations(element: string): CSSStyleDeclaration[] | undefined {
+    const styleSheet = this.getStyleSheet();
+    const styleRule = Array.from(styleSheet.cssRules)?.filter(
+      ele => ele.type === CSSTypes.style
+    ) as CSSStyleRule[] | undefined;
+    return styleRule
+      ?.filter(ele => ele?.selectorText === element)
+      .map(x => x.style);
+  }
+  getStyleDeclaration(element: string): CSSStyleDeclaration | undefined {
+    const styleSheet = this.getStyleSheet();
+    const styleRule = Array.from(styleSheet.cssRules)?.filter(
+      ele => ele.type === CSSTypes.style
+    ) as CSSStyleRule[] | undefined;
+    return styleRule?.find(ele => ele?.selectorText === element)?.style;
+  }
+  getStyleRule(selector: string): CSSStyleRule | undefined {
+    const styleSheet = this.getStyleSheet();
+    const styleRule = Array.from(styleSheet.cssRules)?.filter(
+      ele => ele.type === CSSTypes.style
+    ) as CSSStyleRule[] | undefined;
+    return styleRule?.find(ele => ele?.selectorText === selector);
+  }
+  getCSSRules(element?: string): CSSRule[] | undefined {
+    const styleSheet = this.getStyleSheet();
+    const cssRules = Array.from(styleSheet.cssRules);
+    switch (element) {
+      case 'media':
+        return cssRules?.filter(ele => ele.type === CSSTypes.media);
+      case 'fontface':
+        return cssRules?.filter(ele => ele.type === CSSTypes.fontface);
+      case 'import':
+        return cssRules?.filter(ele => ele.type === CSSTypes.import);
+      case 'keyframes':
+        return cssRules?.filter(ele => ele.type === CSSTypes.keyframes);
+      default:
+        return cssRules;
+    }
+  }
+  isPropertyUsed(property: string): boolean {
+    const styleSheet = this.getStyleSheet();
+    const cssStyleRules = Array.from(styleSheet.cssRules).filter(
+      ele => ele.type === CSSTypes.style
+    ) as CSSStyleRule[];
+    // @ts-ignore
+    return cssStyleRules.some(ele => ele?.style?.[property]);
+  }
+  getRuleListsWithinMedia(conditionText: string): CSSStyleRule[] {
+    const medias = this.getCSSRules('media') as CSSMediaRule[] | undefined;
+    const cond = medias?.find(x => x.conditionText === conditionText) as
+      | CSSMediaRule
+      | undefined;
```

Comment:
```suggestion
    const cond = medias?.find(x => x.conditionText === conditionText)
```
Any particular reason we need this?

Reply:
I cannot remember, right now, if it is entirely needed. A lot of my current typing is to remind me without needing to hover (wait for intellisense)